drafts/taking-a-byte-of-the-carbon-cake.md

```diff
@@ -0,0 +1,291 @@
+That said there are few metrics available from cloud providers to capture transfers within their network, and thus within your system’s network.
+
+
+## What can we do?  What 'byte' can we take?
+
+So to reduce the carbon footprint of the software we create, what ‘byte’ can we take of the carbon cake?
+
+First, there are some excellent best practices available on sites such as [Patterns, Green Software Foundation](https://patterns.greensoftware.foundation/)
+and I recommend a read through the patterns. Amazon Cloud (AWS) also have valuable guidance in their AWS Well-architected Sustainability.  Google Cloud (GCP) also has good credentials for transparency in how they calculate the carbon footprint of their services.
+
+With an engineering mindset, it’s tempting to focus on the challenge of improvements to compute workloads; 
+we are trained to optimise and it’s arguably one of the most challenging and fun aspects. 
+Key patterns in this space are 
+demand-shaping (doing work at a rate to match the current carbon-free energy availability in the region), 
+demand-shifting  (moving workloads to a region with lower carbon energy cost) and 
+time-shifting (deferring until a time when carbon-free energy is available). There’s already some useful tools in this space, 
+such as a Kubernetes (carbon-aware) scheduler.
+
+However with the trend in cloud providers towards fully-green energy, it’s likely that their data centres in the most popular regions will be 100% soon, and so the gains in moving workloads from one to another are decreasing. This diminishes the benefits of demand shifting and demand shaping, leaving time-shifting as a reasonable option.
+
+Therefore we purposefully ignore demand-shaping and demand-shifting in the following opinionated view of the activities with the most significant impact; 
+focusing instead on three groups; easy-wins, medium effort and those activities at architecture design time:
+
+
+### First: Low-Hanging Fruit
+
+![Low hanging fruit](https://cdn.hashnode.com/res/hashnode/image/upload/v1730804092300/ebrrwNFL7.jpeg?auto=format)
+
+* **Measure**
+
+  Start with carbon footprint tools, which will provide a ball-park figure of carbon usage decomposed over services. Be wary that these may not include ‘free usage’ (where free means cost-free of course) and may lack accuracy on shared ‘serverless’ services, eg: lambda functions.  Third party data from CCF or ClimateIQ data provides more granular detail.
+
+
+* **Switch off**
+
+  Particularly relevant for dev and test envs, but also look for resources or services no longer actively used, for example S3 buckets, old logs. 
+  See [GSF pattern: delete unused storage](https://patterns.greensoftware.foundation/catalog/cloud/delete-unused-storage-resources)
+
+
+* **Scale appropriately**
+
+  Ramp down for quiet business periods.  Ramp down environments or services as appropriate.  Consider containerising or co-hosting services where CPU usage is low, such as 
```

Comment:
switching off during night can also be mentioned here, as in "ramp down (even to zero)..."